src/ion/liveranges.rs

```diff
@@ -509,8 +505,8 @@ impl<'a, F: Function> Env<'a, F> {
 
                         // If both src and dest are pinned, emit the
                         // move right here, right now.
-                        if self.vregs[src.vreg().vreg()].is_pinned
-                            && self.vregs[dst.vreg().vreg()].is_pinned
+                        if self.func.is_pinned_vreg(src.vreg()).is_some()
+                            && self.func.is_pinned_vreg(dst.vreg()).is_some()
```

Comment:
This would look better as a `match`:
```rust
match (self.func.is_pinned_vreg(src.vreg()), self.func.is_pinned_vreg(dst.vreg())) {
    (Some(src_preg), Some(dst_preg)) => { ... }
    (Some(preg), None) | (None, Some(preg))  => { ... }
    (None, None) => { ... }
}
```

Reply:
Done!